tracpro/baseline/models.py

```diff
@@ -68,15 +68,23 @@ def _get_answers(self, question, regions):
             question=question, submitted_on__gte=start, submitted_on__lt=end)
         answers = answers.annotate(region_name=F('response__contact__region__name'))
         answers = answers.select_related('response', 'response__contact')
+
+        all_regions = answers.values('response__contact__region',
+                                     'region_name').distinct('response__contact__region')
+
         if regions:
             answers = answers.filter(response__contact__region__in=regions)
-        return answers
+            all_regions = answers.values('response__contact__region',
```

Comment:
The eventual use of this in the template is kind of confusing. What about changing this to something like: `all_regions = answers.values_list('response__contact__region', 'region_name')`

Then you could use something like this in the template: `{% for region_id, region_name in all_regions %}`